### PR TITLE
wip: sync strategy for summaries

### PR DESCRIPTION
**Pull request by devjoaov**
## Summary by CodeRabbit

- **New Features**
  - Enhanced data model to focus on topics instead of forum posts.
  - Introduced functionality for scheduled tasks and periodic synchronization.
  - Added new service for managing topic summarization.
  - Implemented flexible data retrieval based on summarization status.

- **Bug Fixes**
  - Improved methods for saving and updating topic data.
  - Enhanced error handling and data retrieval processes.

- **Chores**
  - Simplified migration processes and updated existing database structures.
  - Removed unused dependencies and updated project requirements.

### CODE REVIEW

File: pkg/op-brains/op_brains/documents/optimism.py [hunk 1]
```diff
@@ -82,10 +82,14 @@ def get_db_name(self) -> str:
...
-        out_db = connect_db.retrieve_data(
-            f'select "rawData", url, type, "externalId" from "{RAW_FORUM_DB}"'
-        )
+    def retrieve(only_not_summarized: bool = False):
+        if only_not_summarized:
+            query = f'select "rawData", url, type, "externalId" from "{RAW_FORUM_DB}" where "needsSummarize"=True'
+        else:
+            query = f'select "rawData", url, type, "externalId" from "{RAW_FORUM_DB}"'
+
+        out_db = connect_db.retrieve_data(query)
```
Comment (devjoaov) [inline pkg/op-brains/op_brains/documents/optimism.py:91]: https://linear.app/bleu-llc/issue/OP-99/refactor-op-brains-to-use-models-from-op-data-db

File: pkg/op-brains/op_brains/summarizer/summarize_out_of_date_threads.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
+import os
```
Comment (devjoaov) [inline pkg/op-brains/op_brains/summarizer/summarize_out_of_date_threads.py:1]: this file was inspired by `pkg/op-brains/op_brains/summarizer/summarize_all_threads.py`, the difference is that we take only the unsummarized topics, summarize them and then save the summaries in the database

File: pkg/op-data/Makefile [hunk 3]
```diff
@@ -9,7 +9,7 @@ seed_categories:
 seed_topics:
 	@poetry install
 	@poetry run sync raw_topics
-	@poetry run sync topics
+	# @poetry run sync topics
```
Comment (ribeirojose) [inline pkg/op-data/Makefile:12]: 👀

File: pkg/op-brains/op_brains/documents/optimism.py (hunk 1 above)
Comment (ribeirojose) [reply pkg/op-brains/op_brains/documents/optimism.py:91]: think we could leave this as is if it's working 🤷

**Pull request merged**